src/main/java/com/traveltime/sdk/TravelTimeSDK.java

```diff
@@ -99,6 +99,17 @@
             .mapLeft(cause -> new IOError(cause, IO_CONNECTION_ERROR + cause.getMessage()));
     }
 
+    public <T> CompletableFuture<Either<TravelTimeError, T>> sendProtoAsync(ProtoRequest<T> request) {
+        final CompletableFuture<Either<TravelTimeError, T>> future = new CompletableFuture<>();
```

Comment:
I'm not sure this is what you want - semantically this will create a future, sequentially perform a request which completes the future and returns the future.

What you want is create & return a future and asynchronously update it using CompletableFuture.supplyAsync https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#supplyAsync-java.util.function.Supplier-java.util.concurrent.Executor- Probably the most flexible way would be to expose a method that accepts an executor and another that uses the default one

Reply:
I don't understand what's the problem with my current approach. I took it from here https://www.baeldung.com/guide-to-okhttp . I construct request, enqueue it and than wait when it will be readable

Reply:
when I return the future it's not completed yet.

Reply:
Yeah sorry that should be fine - didn't understood what the code did - I though the error on peakLeft was after IO - not for creating the request itself.